File: src/org/jsmol/test/JSTest.java
/* $RCSfile$
 * $Author: hansonr $
 * $Date: 2012-05-19 09:56:08 -0500 (Sat, 19 May 2012) $
 * $Revision: 17160 $
 *
 * Copyright (C) 2000-2005  The Jmol Development Team
 *
 * Contact: jmol-developers@lists.sf.net
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
package org.jsmol.test;

import org.jmol.util.P3;

import org.jmol.util.Logger;
import org.jmol.viewer.JC;

public class JSTest extends JSmol {

	public static void main(String[] args) {	
    JSTest jmolApp = new JSTest(args);
    jmolApp.sayHello("...second time...");
    jmolApp = new JSTest(args);
    jmolApp.sayHello("...third time...");
    jmolApp = new JSTest(args);
  }
	
	private static P3 pt2 = P3.new3(2,3,4);
  
	private JSTest(String[] args) {
  	super();
  	System.out.println("testing123");
  	sayHello("JSTest constructor -- after super()");
  	testStatic();
		bs.set((int)(12 + pt2.x));
		pt.x = -1;
  	sayHello("OK-JSTest pt2.x = " + pt2);
		pt2.x -= 1;
		TestInner ti = new TestInner();
		ti.say("ok - ti");
		Logger.info("test log info");
		Logger.debug("test log debug");
		Logger.error("test log error");
		Logger.info(testArray[3]);
		Logger.info(JC.getShapeClassName(JC.SHAPE_HOVER, false));
	}
	
	static String[] testArray = {"a", "b", "c", "d"};
	class TestInner {
		void say(String msg) {
			char a = '@';
			int x = 2 + a;
			int y = 2;
			String x1 = "3" + a;
			char x2 = (char) (a + y);
      System.out.println("x,x1,x2=" + x + "," + x1 + "," + x2);			
			sayHello(checkMap('K'));
			sayHello(myTest("test"));
			sayHello(checkMap((String) getObj("string")));
			sayHello(checkMap(((Boolean) getObj("boolean")).booleanValue()));
//			sayHello(checkMap(true));
			sayHello(msg);
		}
		String myTest(String s) { 
			return s; 
		}
		String checkMap(char c) {
			return "OK-checkMapchar-" + c;
		}
		String checkMap(String s) {
			return "OK-checkMapchar-" + s;
		}
		String checkMap(boolean b) {
			return "OK-checkMapbool-" + b;
		}
		Object getObj(String type) {
			if (type.equalsIgnoreCase("string"))
				return "testing";
			return Boolean.TRUE;
		}
	}

}

File: js/org/jsmol/test/JSTest.js
Clazz.declarePackage ("org.jsmol.test");
Clazz.load (["org.jsmol.test.JSmol", "javax.vecmath.Point3f"], "org.jsmol.test.JSTest", ["java.lang.Boolean", "org.jmol.util.Logger", "org.jmol.viewer.JmolConstants"], function () {
c$ = Clazz.decorateAsClass (function () {
if (!Clazz.isClassDefined ("org.jsmol.test.JSTest.TestInner")) {
org.jsmol.test.JSTest.$JSTest$TestInner$ ();
}
Clazz.instantialize (this, arguments);
}, org.jsmol.test, "JSTest", org.jsmol.test.JSmol);
c$.main = Clazz.defineMethod (c$, "main", 
function (args) {
var jmolApp =  new org.jsmol.test.JSTest (args);
jmolApp.sayHello ("...second time...");
jmolApp =  new org.jsmol.test.JSTest (args);
jmolApp.sayHello ("...third time...");
jmolApp =  new org.jsmol.test.JSTest (args);
}, "~A");
Clazz.makeConstructor (c$, 
($fz = function (args) {
Clazz.superConstructor (this, org.jsmol.test.JSTest);
System.out.println ("testing123");
this.sayHello ("JSTest constructor -- after super()");
this.testStatic ();
org.jsmol.test.JSmol.bs.set (Math.round ((12 + org.jsmol.test.JSTest.pt2.x)));
org.jsmol.test.JSmol.pt.x = -1;
this.sayHello ("OK-JSTest pt2.x = " + org.jsmol.test.JSTest.pt2);
org.jsmol.test.JSTest.pt2.x -= 1;
var ti = Clazz.innerTypeInstance (org.jsmol.test.JSTest.TestInner, this, null);
ti.say ("ok - ti");
org.jmol.util.Logger.info ("test log info");
org.jmol.util.Logger.debug ("test log debug");
org.jmol.util.Logger.error ("test log error");
org.jmol.util.Logger.info (org.jsmol.test.JSTest.testArray[3]);
org.jmol.util.Logger.info (org.jmol.viewer.JmolConstants.getShapeClassName (33, false));
}, $fz.isPrivate = true, $fz), "~A");
c$.$JSTest$TestInner$ = function () {
Clazz.pu$h ();
c$ = Clazz.decorateAsClass (function () {
Clazz.prepareCallback (this, arguments);
Clazz.instantialize (this, arguments);
}, org.jsmol.test.JSTest, "TestInner");
Clazz.defineMethod (c$, "say", 
function (a) {
var b = '@';
var c = 2 + b.charCodeAt (0);
var d = 2;
var e = "3" + b;
var f = String.fromCharCode ((b.charCodeAt (0) + d));
System.out.println ("x,x1,x2=" + c + "," + e + "," + f);
this.b$["org.jsmol.test.JSTest"].sayHello (this.checkMap ('K'));
this.b$["org.jsmol.test.JSTest"].sayHello (this.myTest ("test"));
this.b$["org.jsmol.test.JSTest"].sayHello (this.checkMap (this.getObj ("string")));
this.b$["org.jsmol.test.JSTest"].sayHello (this.checkMap ((this.getObj ("boolean")).booleanValue ()));
this.b$["org.jsmol.test.JSTest"].sayHello (a);
}, "~S");
Clazz.defineMethod (c$, "myTest", 
function (a) {
return a;
}, "~S");
Clazz.defineMethod (c$, "checkMap", 
function (a) {
return "OK-checkMapchar-" + a;
}, "~S");
Clazz.defineMethod (c$, "checkMap", 
function (a) {
return "OK-checkMapchar-" + a;
}, "~S");
Clazz.defineMethod (c$, "checkMap", 
function (a) {
return "OK-checkMapbool-" + a;
}, "~B");
Clazz.defineMethod (c$, "getObj", 
function (a) {
if (a.equalsIgnoreCase ("string")) return "testing";
return Boolean.TRUE;
}, "~S");
c$ = Clazz.p0p ();
};
c$.pt2 = c$.prototype.pt2 =  new javax.vecmath.Point3f (2, 3, 4);
Clazz.defineStatics (c$,
"testArray", ["a", "b", "c", "d"]);
});
